fe/fe-core/src/main/java/org/apache/doris/analysis/CreateMaterializedViewStmt.java

```diff
@@ -193,6 +199,15 @@ public void analyzeSelectClause() throws AnalysisException {
                 // Function must match pattern.
                 FunctionCallExpr functionCallExpr = (FunctionCallExpr) selectListItemExpr;
                 String functionName = functionCallExpr.getFnName().getFunction();
+                // current version not support count(distinct) function in creating materialized view
+                if (!isReplay) {
+                    for (Map.Entry<String, MVColumnPattern> env : FN_NAME_TO_PATTERN.entrySet()){
+                        if (!env.getValue().match(functionCallExpr)){
```

Comment:
Each function corresponds to its own pattern. He only needs to ensure that his pattern is passed. It is not necessary to satisfy all patterns.